azbus/receiver.go

```diff
@@ -302,12 +348,86 @@ func (r *Receiver) ReceiveMessages(handler Handler) error {
 
 }
 
+func (r *Receiver) receiveMessagesInParallel() error {
```

Comment:
Is it possible to unit test this function?

Waitgroups and go routines are tricky and its worth making sure we have unit tests to ensure they are doing what we intend.